File: Backend/HRMS/HRMS.Application/Features/Personnel/Employees/DTOs/CreateEmployeeDto.cs
using System;

namespace HRMS.Application.Features.Personnel.Employees.DTOs
{
    /// <summary>
    /// كائن نقل البيانات لإنشاء موظف جديد
    /// </summary>
    /// <remarks>
    /// يُستخدم لاستقبال بيانات الموظف الجديد من العميل عبر الـ API.
    /// يتم التحقق من صحة البيانات باستخدام FluentValidation قبل المعالجة.
    /// </remarks>
    public class CreateEmployeeDto
    {
        /// <summary>
        /// الرقم الوظيفي (يجب أن يكون فريداً)
        /// </summary>
        public string EmployeeNumber { get; set; }

        /// <summary>
        /// الاسم الأول بالعربية (مطلوب)
        /// </summary>
        public string FirstNameAr { get; set; }

        /// <summary>
        /// الاسم الثاني (اسم الأب) بالعربية
        /// </summary>
        public string SecondNameAr { get; set; }

        /// <summary>
        /// الاسم الثالث (اسم الجد) بالعربية
        /// </summary>
        public string ThirdNameAr { get; set; }

        /// <summary>
        /// اللقب (اسم العائلة) بالعربية (مطلوب)
        /// </summary>
        public string HijriLastNameAr { get; set; }

        /// <summary>
        /// الاسم الكامل بالإنجليزية كما في الجواز
        /// </summary>
        public string FullNameEn { get; set; }

        /// <summary>
        /// الجنس (M: ذكر، F: أنثى)
        /// </summary>
        public string Gender { get; set; }

        /// <summary>
        /// تاريخ الميلاد
        /// </summary>
        public DateTime BirthDate { get; set; }

        /// <summary>
        /// الحالة الاجتماعية (SINGLE, MARRIED, DIVORCED, WIDOWED)
        /// </summary>
        public string MaritalStatus { get; set; }

        /// <summary>
        /// معرف الجنسية (مطلوب)
        /// </summary>
        public int NationalityId { get; set; }

        /// <summary>
        /// معرف الوظيفة (مطلوب)
        /// </summary>
        public int JobId { get; set; }

        /// <summary>
        /// معرف القسم (مطلوب)
        /// </summary>
        public int DeptId { get; set; }

        /// <summary>
        /// معرف المدير المباشر (اختياري)
        /// </summary>
        public int? ManagerId { get; set; }

        /// <summary>
        /// تاريخ المباشرة في العمل
        /// </summary>
        public DateTime JoiningDate { get; set; }

        /// <summary>
        /// البريد الإلكتروني الرسمي
        /// </summary>
        public string Email { get; set; }

        /// <summary>
        /// رقم الهاتف الجوال
        /// </summary>
        public string Mobile { get; set; }
    }
}

File: Frontend/HRMS_Frontend/src/app/features/personnel/models/create-employee.dto.ts
import { 
    Address, 
    BankAccount, 
    Certification, 
    Contract, 
    Dependent, 
    EmergencyContact, 
    EmployeeDocument, 
    Experience, 
    Qualification 
} from './sub-models';

export interface CreateEmployeeDto {
    // Personal Info
    employeeNumber: string;
    firstNameAr: string;
    secondNameAr: string;
    thirdNameAr: string;
    lastNameAr: string;
    fullNameEn: string;
    birthDate: Date | string;
    gender: 'Male' | 'Female' | string;
    mobile: string;
    email?: string;
    nationalityId: number;
    nationalId: string;
    maritalStatus?: string; // Found in one of the POST schemas
    
    // Employment Info
    departmentId: number;
    jobId: number;
    managerId?: number;
    hireDate: Date | string;
    
    // Professional Info
    licenseNumber?: string;
    licenseExpiryDate?: Date | string;
    specialty?: string;
    
    // Financial Info (Direct fields)
    basicSalary: number;
    housingAllowance: number;
    transportAllowance: number;
    medicalAllowance: number;
    
    // Bank Info (Direct fields logic often redundant with BankAccounts array, but present in schema)
    bankId?: number;
    ibanNumber?: string;

    // Nested Lists
    qualifications: Qualification[];
    experiences: Experience[];
    emergencyContacts: EmergencyContact[];
    contracts: Contract[];
    certifications: Certification[];
    bankAccounts: BankAccount[];
    dependents: Dependent[];
    addresses: Address[];
    documents: EmployeeDocument[];
}
